### ISSUE TITLE
Unique reactions does not update messages properly

### ISSUE THREAD

**Issue by remvst** [GetStream/stream-chat-swift#2383]
## What did you do?

Override `reactionPickerVC`:
```swift
        components.reactionPickerVC = MyChatMessageReactionsPickerViewController.self
```

```swift
class MyChatMessageReactionsPickerViewController: ChatMessageReactionsPickerVC {
    /// Copy/pasted implementation of ChatMessageReactionsPickerVC's, adding enforceUnique
    override func toggleReaction(_ reaction: MessageReactionType) {
        guard let message = messageController.message else { return }

        let completion: (Error?) -> Void = { [weak self] _ in
            self?.dismiss(animated: true)
        }

        let shouldRemove = message.currentUserReactions.contains { $0.type == reaction }
        shouldRemove
            ? messageController.deleteReaction(reaction, completion: completion)
            : messageController.addReaction(reaction, enforceUnique: true, completion: completion)
    }
}
```

## What did you expect to happen?

Reactions should be unique on both messages and the picker view.

## What happened instead?

Message in message list looks fine 👍 
![Screen Shot 2022-11-21 at 10 30 58 AM](https://user-images.githubusercontent.com/3472387/203094138-5fef448a-3b42-47ff-8217-8c46ea1ea5b0.png)

When long pressing again, multiple reactions are still selected 👎 
![Screen Shot 2022-11-21 at 10 31 24 AM](https://user-images.githubusercontent.com/3472387/203094238-c8cb8610-d111-47e5-9a94-36b265cb896b.png)

It seems to fix itself after restarting the app.

## GetStream Environment
**GetStream Chat version:** 4.24.0
**GetStream Chat frameworks:** StreamChat, StreamChatUI
**iOS version:** 15
**Swift version:** 4
**Xcode version:** 14.1
**Device:** iPhone X

## Additional context

I suspect this is happening because of this function:

```
func addReaction(
        _ type: MessageReactionType,
        score: Int,
        enforceUnique: Bool,
        extraData: [String: RawJSON],
        messageId: MessageId,
        completion: ((Error?) -> Void)? = nil
    ) {
        let version = UUID().uuidString

        let endpoint: Endpoint<EmptyResponse> = .addReaction(
            type,
            score: score,
            enforceUnique: enforceUnique,
            extraData: extraData,
            messageId: messageId
        )

        database.write { session in
            do {
                let reaction = try session.addReaction( // <--- this line right here doesn't look at enforceUnique
                    to: messageId,
                    type: type,
                    score: score,
                    extraData: extraData,
                    localState: .sending
                )
                reaction.version = version
            } catch {
                log.warning("Failed to optimistically add the reaction to the database: \(error)")
            }
        } completion: { [weak self, weak repository] error in
            self?.apiClient.request(endpoint: endpoint) { result in
                guard let error = result.error else { return }

                if self?.canKeepReactionState(for: error) == true { return }

                repository?.undoReactionAddition(on: messageId, type: type)
            }
            completion?(error)
        }
    }
```

**Comment by nuno-vieira** [comment]
Hi @remvst!

Thank you for the report! I can confirm I can reproduce this issue. And you also seem right, the local database doesn't look at the `enforceUnique` value, so most likely the problem is from there, thank you for all the additional details.

We will tackle this one as soon as possible and we will let you know once we have it fixed.

Best,
Nuno

**Comment by nuno-vieira** [comment]
> > Hi @remvst!
> 
> > 
> 
> > Thank you for the report! I can confirm I can reproduce this issue. And you also seem right, the local database doesn't look at the `enforceUnique` value, so most likely the problem is from there, thank you for all the additional details.
> 
> > 
> 
> > We will tackle this one as soon as possible and we will let you know once we have it fixed.
> 
> > 
> 
> > Best, Nuno
> 
> 
> 
> Do you think it would also be possible to add unique reactions as a general setting? (i.e. not having to implement our own `toggleReaction()`)

We will keep that in mind when we pick this one up 👍 Makes total sense

**Comment by nuno-vieira** [comment]
This fix is now available in the 4.26.0 release.

Thank you for your patience.

Best,
Nuno
